framework/auth/views.py

```diff
@@ -49,26 +49,33 @@ def reset_password(auth, **kwargs):
         }
         raise HTTPError(400, data=error_data)
 
-    if request.method == 'POST' and form.validate():
-        # new random verification key, allows CAS to authenticate the user w/o password one time only.
-        user_obj.verification_key = security.random_string(20)
-        user_obj.set_password(form.password.data)
-        user_obj.save()
-        status.push_status_message('Password reset', kind='success', trust=False)
-        # Redirect to CAS and authenticate the user with a verification key.
-        return redirect(cas.get_login_url(
-            web_url_for('user_account', _absolute=True),
-            auto=True,
-            username=user_obj.username,
-            verification_key=user_obj.verification_key
-        ))
-
-    forms.push_errors_to_status(form.errors)
     return {
         'verification_key': verification_key,
     }
 
 
+def reset_password_post(verification_key):
+    """ attempt to reset password if request form passed
+    """
+    user_obj = get_user(verification_key=verification_key)
+    if not user_obj:
+        error_data = {
+            'message_short': 'Invalid url.',
```

Comment:
+1 for custom error messages.

Super minor: Remove the period and uppercase "URL".